src/dependency-manager/src/configs/service-config.ts

```diff
@@ -0,0 +1,292 @@
+    let debug_path = this.getDebugPath();
+    if (debug_path) {
+      const debug_path_lstat = fs.lstatSync(debug_path);
+      if (debug_path_lstat.isFile()) {
+        debug_path = path.dirname(debug_path);
+      }
+    }
+
+    let build_config = this.getBuildConfig();
+
+    if (!debug_path && !build_config || this.getImage()) {
+      return undefined;
+    }
+
+    build_config = build_config || {};
+    if (build_config.context && debug_path) {
+      build_config.context = path.join(debug_path, build_config.context);
+    } else if (debug_path) {
+      build_config.context = debug_path;
+    }
+
+    if (build_config.dockerfile && debug_path) {
+      build_config.dockerfile = path.join(debug_path, build_config.dockerfile);
+    }
+
+    return build_config;
+  }
+
+  /**
+   * Get either the debug or default command to run
+   * @param debug [boolean] (default: false)
+   */
+  public getPrioritizedCommand(debug = false) {
+    if (debug) {
+      return this.getDebugCommand() || this.getCommand();
+    }
+
+    return this.getCommand();
+  }
+
+  /**
+   * Get either the debug or default entrypoint to run
+   * @param debug [boolean] (default: false)
+   */
+  public getPrioritizedEntrypoint(debug = false) {
+    if (debug) {
+      return this.getDebugEntrypoint() || this.getEntrypoint();
+    }
+
+    return this.getEntrypoint();
+  }
+
+  public merge(config: BaseServiceConfig) {
+    this.setRef(config.getRef() || this.getRef());
+    this.setName(config.getName() || this.getName());
+    this.setMetadata({
+      ...this.getMetadata(),
+      ...config.getMetadata(),
+    });
+    this.setInterfaces(new Map([
+      ...this.getInterfaces(),
+      ...config.getInterfaces(),
+    ]));
+    this.setParameters(new Map([
+      ...this.getParameters(),
+      ...config.getParameters(),
+    ]));
+    this.setVolumes(new Map([
+      ...this.getVolumes(),
+      ...config.getVolumes(),
+    ]));
+    this.setDependencies([
+      ...this.getDependencies(),
+      ...config.getDependencies(),
+    ]);
+    this.setCommand(config.getCommand() || this.getCommand());
+    this.setEntrypoint(config.getEntrypoint() || this.getEntrypoint());
+    this.setImage(config.getImage() || this.getImage());
+    this.setBuildConfig(config.getBuildConfig() || this.getBuildConfig());
+    this.setLivenessProbe({
+      ...this.getLivenessProbe(),
+      ...config.getLivenessProbe(),
+    });
+    this.setNotifications(new Map([
+      ...this.getNotifications(),
+      ...config.getNotifications(),
+    ]));
+    this.setSubscriptions(new Map([
+      ...this.getSubscriptions(),
+      ...config.getSubscriptions(),
+    ]));
+    this.setPlatformsConfig({
+      ...this.getPlatformsConfig(),
+      ...config.getPlatformsConfig(),
+    });
+    this.setDebugPath(config.getDebugPath() || this.getDebugPath());
+    this.setDebugCommand(config.getDebugCommand() || this.getDebugCommand());
```

Comment:
We should be treating the debug block as ServiceConfig and merging it in when debug is enabled. Having to explicitly re-define each option in the service config to override is the problem we were trying to avoid. I'd already done this here: 
https://github.com/architect-team/architect-cli/blob/master/src/dependency-manager/src/manager.ts#L76